tests/test-update-refspec-checkpoint.c

```diff
@@ -1614,10 +1617,13 @@ test_update_refspec_checkpoint_eos3a_eos4 (EosUpdaterFixture *fixture,
                                   &error);
       g_assert_no_error (error);
 
+      test_is_aarch64 = (g_strcmp0 (tests[i].uname_machine, "aarch64") == 0);
+      src_is_eos3a = (g_strcmp0 (tests[i].src_ref, "eos3a") == 0);
       if (tests[i].expect_checkpoint_followed &&
           (!host_is_aarch64 ||
            tests[i].force_follow_checkpoint ||
-           tests[i].uname_machine != NULL))
+           !test_is_aarch64 ||
+           (test_is_aarch64 && !src_is_eos3a)))
```

Comment:
I don’t understand the reasoning for changing the conditions here, or introducing `test_is_aarch64` or `src_is_eos3a`. Could you explain the reasoning please?

Reply:
I was confused by the test's expected result for the aarch64 part when I saw and tuned the test plans at first time.  After you ask this question, I realize the expected result is already controlled by **expect_checkpoint_followed** and I I have already modified it in the test plans.
So, changed the check condition back.